Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM node:24-alpine AS builder
+FROM node:lts-alpine AS builder
```

Comment:
This is a bad practice, please don't do this. Always lock down your dependencies, at least with a major version.

Reply:
@guerzon 
It's not totally bad. as long at it is LTS. but yes, it is better to lock to some version. Not 24 yet, probably around version 22.

Reply:
The stability of the upstream Node distribution is not the reason we lock down dependencies in this case. Rather, we do it to maximise compatibility among developers' tooling. It also helps to minimize bugs caused by mismatched versions across devs, the CI system, and the runtime environment. This is a standard best practice.

Version 24 was set, but it can be adjusted to match what the original dev used, or to whatever most of the devs are using right now.

Reply:
> The stability of the upstream Node distribution is not the reason we lock down dependencies in this case.

I assume the author and me already know about this.

Again, I agree with you. I'm just saying it's not totally bad, as long as it is documented properly. 

> This is a standard best practice.

I believe, there is no such thing as best practice, only agreed or common/standard practice.

Just check the "bests" repos out there, and see if they really share the same best standard practice.

Reply:
> I assume the author and me already know about this.

In that case, can you enlighten us what you were referring here:

> as long at it is LTS

> I believe, there is no such thing as best practice, only agreed or common/standard practice.

Call it whatever you want, there's no point arguing semantics here.

Reply:
> In that case, can you enlighten us what you were referring to with this:

here
> as long as it is documented properly.

Again, I'm saying that I agree with you and I will almost always use your approach, but it's not 100% right to call it bad practice.

Primary benefit of defining lts is we get free "auto update" of node versions. Since we have some automations, any issues with dependencies can be detected.